src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseCountProperly.cs

```diff
@@ -403,9 +405,9 @@ private static void DetermineReportForInvocationAnalysis(OperationAnalysisContex
             if (!shouldReplaceParent)
             {
                 // Parent expression doesn't met the criteria to be replaced; fallback on analysis for standalone call to Count().
-                if (!isAsync && !hasPredicate)
+                if (!isAsync && !hasPredicate && operation is IInvocationOperation invocation)
                 {
-                    AnalyzeCountInvocationOperation(context, (IInvocationOperation)operation);
+                    AnalyzeCountInvocationOperation(context, invocation);
```

Comment:
This change and the one on line 408 are not related to the issue but there was nothing ensuring we couldn't reach this line with a non invocation operation.

Reply:
Since this is only called from `AnalyzeInvocationOperation` which is the method that ensures that `operation` is `IInvocationOperation`, I was assuming this would always be true.

Instead of jumping through IInvocationOperation -> IOperation -> IInvocationOperation, I would rather change the parameter type to IInvocationOperation to avoid this cast but then you will have to use explicit calls on the methods where `operation` is passed as a parameter.